handlers/messagebird/messagebird.go

```diff
@@ -0,0 +1,302 @@
+package messagebird
+
+/*
+ * Handler for MessageBird
+ */
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"strconv"
+
+	"fmt"
+
+	"net/http"
+	"time"
+
+	"github.com/buger/jsonparser"
+	"github.com/golang-jwt/jwt/v5"
+	"github.com/nyaruka/courier"
+	"github.com/nyaruka/courier/handlers"
+	"github.com/nyaruka/gocommon/urns"
+	"github.com/sirupsen/logrus"
+)
+
+var (
+	smsURL                    = "https://rest.messagebird.com/messages"
+	mmsURL                    = "https://rest.messagebird.com/mms"
+	signatureHeader           = "Messagebird-Signature-Jwt"
+	maxRequestBodyBytes int64 = 1024 * 1024
+	// error code messagebird returns when a contact has sent "stop"
+	errorStopped = 103
+)
+
+type Message struct {
+	Recipients []string `json:"recipients"`
+	Reference  string   `json:"reference,omitempty"`
+	Originator string   `json:"originator"`
+	Subject    string   `json:"subject,omitempty"`
+	Body       string   `json:"body,omitempty"`
+	MediaURLs  []string `json:"mediaUrls,omitempty"`
+}
+
+type ReceivedStatus struct {
```

Comment:
We need tests for status reports too

Reply:
working on those

Reply:
We need to specify the field form names too

Reply:
and those that are required to validate the type

Reply:
the status reports are a GET, so the current handler tests don't support it. any workarounds?

Reply:
You just pass the get parameters in the URL and not specify Data field 

Similar to 
https://github.com/nyaruka/courier/blob/ece968bc2c5b762b711adf14ebd158a95c95d316/handlers/external/external_test.go#L250

Reply:
but thats still going to make an http POST and the handler is expecting a GET

Reply:
Better example actually is 

https://github.com/nyaruka/courier/blob/ece968bc2c5b762b711adf14ebd158a95c95d316/handlers/burstsms/burstsms_test.go#L22-L50

Reply:
When Data is `""` or not specified the requests are done with GET

Reply:
Check https://github.com/nyaruka/courier/blob/ece968bc2c5b762b711adf14ebd158a95c95d316/handlers/test.go#L77-L106

Reply:
Line 105 is the one that does the GET requests

Reply:
ahh ok thanks!